lib/src/cluster_manager.dart

```diff
@@ -112,17 +137,52 @@ class ClusterManager<T extends ClusterItem> {
     }
   }
 
+  Future<LatLngBounds> _addPadding(LatLngBounds mapBounds) async {
+    final northEastL = mapBounds.northeast;
+    final southWestL = mapBounds.southwest;
+
+    if (padding != null) {
+      final [northEastC, southWestC] = await Future.wait([
+        GoogleMapsFlutterPlatform.instance
+            .getScreenCoordinate(northEastL, mapId: _mapId!),
+        GoogleMapsFlutterPlatform.instance
+            .getScreenCoordinate(southWestL, mapId: _mapId!),
+      ]);
+
+      final [northEastP, southWestP] = await Future.wait([
+        GoogleMapsFlutterPlatform.instance.getLatLng(
+          northEastC.add(
+            x: padding!.right.toInt(),
+            y: -padding!.top.toInt(),
+          ),
+          mapId: _mapId!,
+        ),
+        GoogleMapsFlutterPlatform.instance.getLatLng(
+          southWestC.add(
+            x: -padding!.left.toInt(),
+            y: padding!.bottom.toInt(),
+          ),
+          mapId: _mapId!,
+        ),
+      ]);
+      return LatLngBounds(southwest: southWestP, northeast: northEastP);
+    }
+    return LatLngBounds(southwest: southWestL, northeast: northEastL);
+  }
+
   /// Retrieve cluster markers
   Future<List<Cluster<T>>> getMarkers() async {
     if (_mapId == null) return List.empty();
 
     final mapBounds = await GoogleMapsFlutterPlatform.instance.getVisibleRegion(mapId: _mapId!);
 
+    final paddedBounds = await _addPadding(mapBounds);
+
     late LatLngBounds inflatedBounds;
     if (clusterAlgorithm == ClusterAlgorithm.geoHash) {
-      inflatedBounds = _inflateBounds(mapBounds);
+      inflatedBounds = _inflateBounds(paddedBounds);
     } else {
-      inflatedBounds = mapBounds;
+      inflatedBounds = paddedBounds;
```

Comment:
for better readability i would suggest using a switch statement.

```dart
    final inflatedBounds = switch (clusterAlgorithm) {
      ClusterAlgorithm.geoHash => _inflateBounds(paddedBounds),
      _ => paddedBounds,
    };
```

Reply:
Fixed in [0d1091c](https://github.com/SebastianKlaiber/google_maps_cluster_manager/pull/3/commits/0d1091cf7bb4f97e274f4258186435bc04d1b4c1)